src/js/utils/decryptVideo.js

```diff
@@ -0,0 +1,143 @@
+ *
+ * Substitute your own keys and KIDs – 32 hex characters each.
+ *
+ * {@link https://ffmpeg.org/}
+ * {@link https://www.bento4.com/documentation/mp4encrypt/}
+ */
+
+/**
+ * KID:key pairs as non-padded base64 values.
+ */
+const KEYS = {
+  x3_uNeUf1hWnuRr8sQkcXg: '7s2ytUnwKnyXzlDBf0lMoA',
+  'BF9-zDWEjtezwBLqdhRCLw': 'mrt6tsxK07hsIZPa2x54bA',
+};
+
+/**
+ * Generates a license for the given key IDs ("kids").
+ *
+ * @param {Uint8Array} message Key session message.
+ * @returns {Uint8Array} License data.
+ */
+const generateLicense = (message) => {
+  const request = JSON.parse(new TextDecoder().decode(message));
+  const keys = [];
+
+  request.kids.forEach((kid) => {
+    keys.push({
+      kty: 'oct',
+      alg: 'A128KW',
+      kid,
+      k: KEYS[kid],
+    });
+  });
+
+  return new TextEncoder().encode(JSON.stringify({
+    keys,
+  }));
+};
+
+/**
+ * Uses Encrypted Media Extensions to decrypt the video.
+ *
+ * @param {HTMLMediaElement} videoElement                    Video element.
+ * @param {object}           encryption                      Encryption data.
+ * @param {string}           encryption.type                 Encryption type.
+ * @param {string}           encryption.mimeCodec            Media MIME type and codec string.
+ * @param {object}           encryption.mediaKeySystemConfig Key system config.
+ * @param {object}           [encryption.key]                Encryption key.
+ * @param {string}           [encryption.key.id]             Key ID as a hexadecimal string.
+ * @param {string}           [encryption.key.value]          Key value as a hexadecimal string.
+ * @returns {void}
+ */
+export default function decryptVideo(videoElement, encryption) {
+  const handleEncrypted = async ({ initDataType, initData }) => {
+    const mediaKeySystemAccess = await navigator.requestMediaKeySystemAccess(
+      encryption.type,
+      [encryption.mediaKeySystemConfig],
+    );
+    const mediaKeys = await mediaKeySystemAccess.createMediaKeys();
+    await videoElement.setMediaKeys(mediaKeys);
+
+    const keySession = mediaKeys.createSession();
+    keySession.addEventListener('message', async (e) => {
+      const license = generateLicense(e.message);
+      await keySession.update(license).catch(
+        /* eslint-disable-next-line no-console */
+        (updateError) => console.error(`keySession.update() failed with error: ${updateError}`),
+      );
+    }, false);
+    await keySession.generateRequest(initDataType, initData);
+  };
+
+  videoElement.addEventListener('encrypted', handleEncrypted, false);
+
+  if ('MediaSource' in window && MediaSource.isTypeSupported(encryption.mimeCodec)) {
+    const mediaSource = new MediaSource();
+    videoElement.src = URL.createObjectURL(mediaSource);
+
+    mediaSource.addEventListener('sourceopen', () => {
+      const sourceBuffer = mediaSource.addSourceBuffer(encryption.mimeCodec);
+
+      fetch(encryption.src).then((response) => {
```

Comment:
Nit: What happens if `fetch` fails?

Reply:
Updated in 281290836b44cdc445dea0ef7410d1082e9d8506, added a `catch` block.